### ISSUE TITLE
SendAsync_SuppressedGlobalStaticPropagationNoListenerAppCtx test failing

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#39691]
This test had been missing a [Theory] attribute and thus wasn't actually running.  Now that that's been fixed, the test is failing 100% of the time, e.g.
````
Unhandled exception. Xunit.Sdk.EqualException: Assert.Equal() Failure
  Expected: True
  Actual:   False
     at Xunit.Assert.Equal[T](T expected, T actual, IEqualityComparer`1 comparer) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\EqualityAsserts.cs:line 41
     at Xunit.Assert.Equal[T](T expected, T actual) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\EqualityAsserts.cs:line 25
     at System.Net.Http.Functional.Tests.DiagnosticsTest.<>c.<SendAsync_SuppressedGlobalStaticPropagationNoListenerAppCtx>b__25_0(String innerSwitchValue, String innerIsInstrumentationEnabled) in D:\repos\corefx\src\System.Net.Http\tests\FunctionalTests\DiagnosticsTests.cs:line 1108
  --- End of stack trace from previous location where exception was thrown ---
     at Microsoft.DotNet.RemoteExecutorHost.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Microsoft.DotNet.RemoteExecutorHost/Program.cs:line 92
      System.Net.Http.Functional.Tests.SocketsHttpHandler_DiagnosticsTest.SendAsync_SuppressedGlobalStaticPropagationNoListenerAppCtx(switchValue: False, isInstrumentationEnabled: True) [FAIL]
        Microsoft.DotNet.RemoteExecutor.RemoteExecutionException : Remote process failed with an unhandled exception.
        Stack Trace:

          Child exception:
            Xunit.Sdk.EqualException: Assert.Equal() Failure
          Expected: True
          Actual:   False
          D:\repos\corefx\src\System.Net.Http\tests\FunctionalTests\DiagnosticsTests.cs(1108,0): at System.Net.Http.Functional.Tests.DiagnosticsTest.<>c.<SendAsync_SuppressedGlobalStaticPropagationNoListenerAppCtx>b__25_0(String innerSwitchValue, String innerIsInstrumentationEnabled)

          Child process:
            System.Net.Http.Functional.Tests, Version=4.2.1.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 System.Net.Http.Functional.Tests.DiagnosticsTest+<>c Int32 <SendAsync_SuppressedGlobalStaticPropagationNoListenerAppCtx>b__25_0(System.String, System.String)

          Child arguments:
            False, True
```
@lmolkova, can you please take a look?

**Comment by karelz** [comment]
@lmolkova is it affecting 3.0? We need to understand the impact ASAP as we are kicking down more and more ...

**Comment by lmolkova** [comment]
this is a test bug and @stephentoub has PR to fix it (thanks @stephentoub)

**Comment by karelz** [comment]
This issue was fixed only in master (5.0) - fixing milestone to 5.0.
@lmolkova @stephentoub can you please confirm it is sufficient and we do not need it / want it in 3.0? Thanks!

**Comment by stephentoub** [comment]
> can you please confirm it is sufficient and we do not need it / want it in 3.0?

The test in release/3.0 is missing a [Theory] attribute, so it's not running at all, so there's nothing that need be done there to fix failures, since there aren't any.  We could choose to enable the test by adding [Theory] and then fixing it as was done here, but it's certainly not required.
